FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_plain_text/DemoTasks/PlaintextMQTTExample.c

```diff
@@ -0,0 +1,656 @@
+
+static void prvMQTTProcessResponse( MQTTPacketInfo_t * pxIncomingPacket,
+									uint16_t usPacketId )
+{
+	switch( pxIncomingPacket->type )
+	{
+		case MQTT_PACKET_TYPE_SUBACK:
+			LogInfo( ( "Subscribed to the topic %s.\r\n", mqttexampleTOPIC ) );
+			/* Make sure ACK packet identifier matches with Request packet identifier. */
+			configASSERT( usSubscribePacketIdentifier == usPacketId );
+			break;
+
+		case MQTT_PACKET_TYPE_UNSUBACK:
+			LogInfo( ( "Unsubscribed from the topic %s.\r\n", mqttexampleTOPIC ) );
+			/* Make sure ACK packet identifier matches with Request packet identifier. */
+			configASSERT( usUnsubscribePacketIdentifier == usPacketId );
+			break;
+
+		case MQTT_PACKET_TYPE_PINGRESP:
+			LogInfo( ( "Ping Response successfully received.\r\n" ) );
+			break;
+
+		/* Any other packet type is invalid. */
+		default:
+			LogWarn( ( "prvMQTTProcessResponse() called with unknown packet type:(%02X).\r\n",
+					   pxIncomingPacket->type ) );
+	}
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvMQTTProcessIncomingPublish( MQTTPublishInfo_t * pxPublishInfo )
+{
+	configASSERT( pxPublishInfo != NULL );
+
+	/* Process incoming Publish. */
+	LogInfo( ( "Incoming QOS : %d\n", pxPublishInfo->qos ) );
+
+	/* Verify the received publish is for the we have subscribed to. */
+	if( ( pxPublishInfo->topicNameLength == strlen( mqttexampleTOPIC ) ) &&
+		( 0 == strncmp( mqttexampleTOPIC, pxPublishInfo->pTopicName, pxPublishInfo->topicNameLength ) ) )
+	{
+		LogInfo( ( "\r\nIncoming Publish Topic Name: %.*s matches subscribed topic.\r\n"
+				   "Incoming Publish Message : %.*s\r\n",
+				   pxPublishInfo->topicNameLength,
+				   pxPublishInfo->pTopicName,
+				   pxPublishInfo->payloadLength,
+				   pxPublishInfo->pPayload ) );
+	}
+	else
+	{
+		LogInfo( ( "Incoming Publish Topic Name: %.*s does not match subscribed topic.\r\n",
+				   pxPublishInfo->topicNameLength,
+				   pxPublishInfo->pTopicName ) );
+	}
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvEventCallback( MQTTContext_t * pxMQTTContext,
+							  MQTTPacketInfo_t * pxPacketInfo,
+							  uint16_t usPacketIdentifier,
+							  MQTTPublishInfo_t * pxPublishInfo )
+{
+	if( ( pxPacketInfo->type & 0xF0U ) == MQTT_PACKET_TYPE_PUBLISH )
+	{
+		prvMQTTProcessIncomingPublish( pxPublishInfo );
+	}
+	else
+	{
+		prvMQTTProcessResponse( pxPacketInfo, usPacketIdentifier );
+	}
+}
+
+/*-----------------------------------------------------------*/
+
+static uint32_t prvGetTimeMs( void )
+{
+	TimeOut_t xCurrentTime = { 0 };
+
+	/* This must be unsigned because the behavior of signed integer overflow is undefined. */
+	uint64_t ullTickCount = 0ULL;
+	uint32_t ulTimeMs = 0UL;
+
+	/* Get the current tick count and overflow count. vTaskSetTimeOutState()
+	 * is used to get these values because they are both static in tasks.c. */
+	vTaskSetTimeOutState( &xCurrentTime );
```

Comment:
This doesn't seem to be the right API to use for getting the time, instead this API is used to adjust the timeout value for the task to be spent in blocked state. Please refer to the documentation here https://www.freertos.org/vTaskSetTimeOutState.html. How about https://www.freertos.org/a00021.html#xTaskGetTickCount?

Reply:
Changed to use xTaskGetTickCount instead